components/ai_chat/renderer/page_text_distilling.cc

```diff
@@ -139,8 +144,29 @@ void AddTextNodesToVector(const ui::AXNode* node,
 
 void DistillPageText(
     content::RenderFrame* render_frame,
+    int32_t global_world_id,
     int32_t isolated_world_id,
     base::OnceCallback<void(const std::optional<std::string>&)> callback) {
+  blink::WebLocalFrame* main_frame = render_frame->GetWebFrame();
+
+  if (ai_chat::features::IsAIHostSpecificDistillationEnabled()) {
+    std::string_view host =
+        url::Origin(((const blink::WebFrame*)main_frame)->GetSecurityOrigin())
```

Comment:
I'm not sure if I'm missing something, but we don't need this cast here, because `blink::WebLocalFrame` is a derived class of `blink::WebFrame`, so you can just call `main->GetSecurityOrigin()`. What do you think?

Reply:
That definitely looks considerably cleaner. This pattern was (mostly/entirely) pre-existing IIRC, so I avoided making any big changes at the time.